File: app/src/main/java/com/univreview/view/contract/ReviewListContract.java
package com.univreview.view.contract;

import com.univreview.fragment.AbsListFragment;
import com.univreview.model.Professor;
import com.univreview.model.Subject;
import com.univreview.model.enumeration.ReviewSearchType;
import com.univreview.view.ReviewTotalScoreView;

import java.util.List;

/**
 * Created by DavidHa on 2017. 7. 14..
 */

public interface ReviewListContract {
    interface View{
        void setDialog(List<String> list);
        Long getSubjectId();
        Long getProfessorId();
        Long getUserId();
        void setResult(int page);
        void setStatus(AbsListFragment.Status status);
        ReviewTotalScoreView getReviewTotalScoreView();
        void setHeaderViewVisibility(boolean isVisibility);
        void setFilterName(String filterName);
    }

    void attachView(View view);
    void detachView();
    void loadReviewItem(ReviewSearchType type, int page);
    void searchProfessor(long subjectId);
    void searchSubject(long professorId);
    List<Subject> getSubjects();
    List<Professor> getProfessors();
}

File: app/src/main/java/com/univreview/view/contract/ReviewListContract.kt
package com.univreview.view.contract

import com.univreview.fragment.AbsListFragment
import com.univreview.listener.OnItemClickListener
import com.univreview.model.enumeration.ReviewSearchType

/**
 * Created by DavidHa on 2017. 8. 6..
 */
interface ReviewListContract {
    interface View : BaseView {
        fun setDialog(list: List<String>, itemClickListener: OnItemClickListener)
        fun setResult(page: Int)
        fun setStatus(status: AbsListFragment.Status)
        fun setFilterName(filterName: String)
    }

    fun loadReviewItem(type: ReviewSearchType, page: Int)
    fun loadFilterList()
}